common/test/acceptance/tests/test_cohorted_courseware.py

```diff
@@ -0,0 +1,226 @@
+"""
+End-to-end test for cohorted courseware. This uses both Studio and LMS.
+"""
+
+from nose.plugins.attrib import attr
+import json
+
+from studio.base_studio_test import ContainerBase
+
+from ..pages.studio.settings_group_configurations import GroupConfigurationsPage
+from ..pages.studio.settings_advanced import AdvancedSettingsPage
+from ..pages.studio.auto_auth import AutoAuthPage as StudioAutoAuthPage
+from ..fixtures.course import XBlockFixtureDesc
+from ..pages.studio.component_editor import ComponentVisibilityEditorView
+from ..pages.lms.instructor_dashboard import InstructorDashboardPage
+from ..pages.lms.course_nav import CourseNavPage
+from ..pages.lms.courseware import CoursewarePage
+from ..pages.lms.auto_auth import AutoAuthPage as LmsAutoAuthPage
+
+from bok_choy.promise import EmptyPromise
+
+
+@attr('shard_1')
+class EndToEndCohortedCoursewareTest(ContainerBase):
+
+    def setUp(self, is_staff=True):
+
+        super(EndToEndCohortedCoursewareTest, self).setUp(is_staff=is_staff)
+        self.staff_user = self.user
+
+        self.content_group_a = "Content Group A"
+        self.content_group_b = "Content Group B"
+
+        # Create a student who will be in "Cohort A"
+        self.cohort_a_student_username = "cohort_a_student"
+        self.cohort_a_student_email = "cohort_a_student@example.com"
+        StudioAutoAuthPage(
+            self.browser, username=self.cohort_a_student_username, email=self.cohort_a_student_email, no_login=True
+        ).visit()
+
+        # Create a student who will be in "Cohort B"
+        self.cohort_b_student_username = "cohort_b_student"
+        self.cohort_b_student_email = "cohort_b_student@example.com"
+        StudioAutoAuthPage(
+            self.browser, username=self.cohort_b_student_username, email=self.cohort_b_student_email, no_login=True
+        ).visit()
+
+        # Create a student who will end up in the default cohort group
+        self.cohort_default_student_username = "cohort default student"
+        self.cohort_default_student_email = "cohort_default_student@example.com"
+        StudioAutoAuthPage(
+            self.browser, username=self.cohort_default_student_username,
+            email=self.cohort_default_student_email, no_login=True
+        ).visit()
```

Comment:
You  could collapse the above three blocks by iterating over `[(self.cohort_a_student_username, self.cohort_a_student_email), (self.cohort_b_student_username, self.cohort_b_student_email), (self.cohort_default_student_username, self.cohort_default_student_email)]`.